ci2/ci/github.py

```diff
@@ -166,6 +180,30 @@ def config(self):
             'sha': self.sha
         }
 
+    def should_build(self):
+        return self.batch is None and (
+            self.most_recent_build is None or self.source_sha != self.most_recent_build.source_sha)
+
+    def build_complete(self):
+        return complete_build_state(self.build_state)
+
+    def should_update_most_recent_build(self):
+        return self.batch is not None and (
+            self.most_recent_build is None or
+            self.build_complete() or
+            (not self.most_recent_build.is_complete() and
+             self.source_sha != self.most_recent_build.source_sha))
```

Comment:
This is tricky!  I don't think we should have two builds running at once.  So we should promote batch to the most recent build if:
 - the batch source_sha matches the current sha, and
 - most recent build is missing or batch is complete.

Then, if we start a new batch and the most recent batch isn't done, we set it to None.